components/engine_traits/src/flush.rs

```diff
@@ -54,6 +54,37 @@ struct FlushProgress {
     last_flushed: [u64; DATA_CFS_LEN],
 }
 
+/// A share state between raftstore and underlying engine.
+///
+/// raftstore will update state changes and corresponding sst apply index, when
+/// apply ingest sst request, it should ensure the sst can be deleted
+/// if the flushed index greater than it .
+#[derive(Debug)]
+pub struct SSTState {
```

Comment:
1) I think sst_applied_index should be per CF, not shared among the CFs.
2) The name of SSTState is also a bit confusing.  Maybe can renamed as SSTApplyIndex

Reply:
In fact, I guess we may have to keep the map of SstMeta and its apply index to make it work.